fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/DruidQueryIdSpec.groovy

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.endpoints
+
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary
+import com.yahoo.bard.webservice.util.JsonSlurper
+import com.yahoo.bard.webservice.util.JsonSortStrategy
+import com.yahoo.bard.webservice.web.filters.BardLoggingFilter
+
+import javax.ws.rs.core.Response
+
+
+class DruidQueryIdSpec extends BaseDataServletComponentSpec {
+    def prefixId = "abcdef"
+    static def testedDruidQuery = false
+
+    @Override
+    def setup() {
+        DimensionDictionary dimensionStore = jtb.configurationLoader.dimensionDictionary
+        dimensionStore.findByApiName("color").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "Foo", "FooDesc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "Bar", "BarDesc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "Baz", "BazDesc"))
+        }
+    }
+
+    @Override
+    Class<?>[] getResourceClasses() {
+        [ DataServlet.class
+        , BardLoggingFilter.class ]
+    }
+
+    @Override
+    String getTarget() {
+        return "data/shapes/week/color"
+    }
+
+    @Override
+    Map<String, List<String>> getQueryParams() {
+        [
+                "metrics": ["width","depth"],
+                "dateTime": ["2014-06-02%2F2014-06-30"],
+                "sort": ["width|desc","depth|asc"]
+        ]
+    }
+
+    @Override
+    boolean compareResult(String result, String expectedResult, JsonSortStrategy sortStrategy = JsonSortStrategy.SORT_MAPS) {
+        if (!testedDruidQuery) {
+            JsonSlurper slurper = new JsonSlurper(sortStrategy)
+            def parsedJson = slurper.parseText(result)
+            testedDruidQuery = true
+            return parsedJson.context.queryId.startsWith(prefixId)
+        }
+        // Only do this for druid test. We're noop'ing everything else. Only testing for queryId prefix.
+        return true
+    }
+
+    @Override
+    String getExpectedDruidQuery() {
+        // Noop
+        """{}"""
+    }
+
+    @Override
+    String getFakeDruidResponse() {
+        // Noop
+        """[]"""
+    }
+
+    @Override
+    String getExpectedApiResponse() {
+        // Noop
+        """{}"""
+    }
+
+    @Override
+    Response makeAbstractRequest(Closure queryParams=this.&getQueryParams) {
```

Comment:
I would appreciate a comment indicating that the only difference between this version of `makeAbstractRequest` and the parent version is that we are adding the `X-Request-ID` to the header. Otherwise, the two implementations are so similar that people may think this implementation is unnecessary, delete it, and then wonder why the test started failing.

Though a better approach may be to add a hook to `makeAbstractRequest`. Basically, add a method `getAdditionalHeaders` that returns headers to add to the request before making the call. We can then have `BaseDataServletComponentSpec::makeAbstractRequest` add the headers from `getAdditionalHeaders` to the request just before making the call.